src/AutoRest.CSharp.V3/Plugins/Modeler.cs

```diff
@@ -201,40 +172,22 @@ private PathSegment[] ToPathParts(string httpRequestUri, Dictionary<string, Cons
             return host.ToArray();
         }
 
-        private int ToStatusCode(StatusCodes arg)
-        {
-            return int.Parse(arg.ToString().Trim('_'));
-        }
+        private static int ToStatusCode(StatusCodes arg) => int.Parse(arg.ToString().Trim('_'));
 
-        private ClientConstant StringConstant(string s)
-        {
-            return new ClientConstant(s, new FrameworkTypeReference(typeof(string)));
-        }
+        private static ClientConstant StringConstant(string s) => new ClientConstant(s, new FrameworkTypeReference(typeof(string)));
 
-        private ClientModel.ClientModel BuildModel(Schema schema)
+        private static ClientModel.ClientModel BuildModel(Schema schema) => schema switch
         {
-            switch (schema)
-            {
-                case SealedChoiceSchema sealedChoiceSchema:
-                    return new ClientEnum(sealedChoiceSchema, sealedChoiceSchema.CSharpName(),
-                        sealedChoiceSchema.Choices.Select(c => new ClientEnumValue(c.CSharpName(), StringConstant(c.Value))))
-                    {
-                        IsStringBased = false
-                    };
-                case ChoiceSchema choiceSchema:
-                    return new ClientEnum(choiceSchema,  choiceSchema.CSharpName(),
-                        choiceSchema.Choices.Select(c => new ClientEnumValue(c.CSharpName(), StringConstant(c.Value))))
-                    {
-                        IsStringBased = true
-                    };
-                case ObjectSchema objectSchema:
-                    return new ClientObject(objectSchema, objectSchema.CSharpName(),
-                        objectSchema.Properties.Where(property => !(property.Schema is ConstantSchema)).Select(CreateProperty),
-                        objectSchema.Properties.Where(property => property.Schema is ConstantSchema).Select(CreateConstant));
-            }
-
-            throw new NotImplementedException();
-        }
+            SealedChoiceSchema sealedChoiceSchema => (ClientModel.ClientModel)new ClientEnum(sealedChoiceSchema,
```

Comment:
This didn't look great but now I feel it's even worse.

Reply:
Addressed. Put these into separate creation methods.